bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/discovery/HomeAssistantDiscovery.java

```diff
@@ -126,6 +130,33 @@ public Set<ThingTypeUID> getSupportedThingTypes() {
         return typeProvider.getThingTypeUIDs();
     }
 
+    /**
+     * Summarize components such as {Switch, Switch, Sensor} into string "Sensor, 2x Switch"
+     *
+     * @param componentNames stream of component names
+     * @return summary string of component names and their counts
+     */
+    static String getComponentNamesSummary(Stream<String> componentNames) {
+        StringBuilder summary = new StringBuilder();
+        @SuppressWarnings("null")
```

Comment:
Can you please avoid that line? 
For what reason did you add it?

Reply:
The warning is 
```
Unsafe null type conversion (type annotations): The value of type 'Collector<@NonNull String,capture#of ?,Map<@NonNull String,@NonNull Long>>' is made accessible using the less-annotated type '@NonNull Collector<? super @NonNull String,Object,@NonNull Map<@NonNull String,Long>>'
```

At least one type annotation difference there seems to be `@NonNull Long` vs `Long`?

Reply:
Resolved with trick in https://github.com/openhab/openhab-addons/pull/13402/commits/ad0a1bf1034b384bcfda003c881acfefd1b3ec1f